src/interceptors/ClientRequest/index.ts

```diff
@@ -104,12 +108,19 @@ export class ClientRequestInterceptor extends Interceptor<HttpRequestEventMap> {
       },
     })
 
+    // Spy on `Header.prototype.set` and `Header.prototype.append` calls
+    // and record the raw header names provided. This is to support
+    // `IncomingMessage.prototype.rawHeaders`.
+    recordRawFetchHeaders()
```

Comment:
I'm enabling the raw headers recording as a part of the ClientRequest interceptor. This is the only place we need it for now.

Reply:
What about unmocked requests?

Reply:
The behavior should be the same for bypassed requests. Let me check tomorrow if I have tests to prove that.

Reply:
@kettanaito Seems like it is. I added a test. If you meant something else, feel free to remove this.
Also, I updated the minimum node version in the package.json to `18.20.0`

Reply:
@kettanaito For some reason, I get this error:
> Exception has occurred: TypeError: Cannot assign to read only property 'Symbol(kRawHeaders)' of object '#<_Headers>'
  at Object.construct (/..../mswjs-interceptors/lib/node/chunk-CFRXZJO4.js:192:39)
    at createResponse (/..../nock/lib/create_response.js:43:20)
    at OverriddenClientRequest.<anonymous> (/.../nock/lib/intercept.js:405:28)
    at OverriddenClientRequest.emit (node:events:530:35)
    at Timeout.respond [as _onTimeout] (/.../nock/lib/playback_interceptor.js:307:11)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)


I'm wondering if I do something wrong:
```js
function createResponse(message: IncomingMessage) {
  ...
  const rawHeaders = new Headers()
  for (let i = 0; i < message.rawHeaders.length; i += 2) {
    rawHeaders.append(message.rawHeaders[i], message.rawHeaders[i + 1])
  }

  const response = new Response(responseBodyOrNull, {
    status: message.statusCode,
    statusText: message.statusMessage || STATUS_CODES[message.statusCode],
    headers: rawHeaders,
  })

  return response
}
```

Reply:
The errors disappear when I set `writable: true` for `kRawHeaders`.
I'm wondering why it works for `interceptors` tests.

Reply:
I also see a problem with the apply and restore functionality. It could be how I implemented it in Nock. I'll continue to investigate this tomorrow.

Reply:
@mikicho, thanks for the tests! I will look at them once I have a moment. 

> Also, I updated the minimum node version in the package.json to 18.20.0

Please, this would be best introduced on its own. Let's not add it as a part of this pull request. 

The "cannot set symbol" error is caused because something tries to set it when it's already set. Interesting, I thought I put guards against that. 

> The errors disappear when I set writable: true for kRawHeaders.

This isn't a solution, it simply means one actor can override previously stored raw headers. This is actually quite dangerous because fetch api primitives _repeatedly construct Headers instances_. 

```js
const h = new Headers()
// ^ This is one headers instance.

const r = new Response(null, { headers: h })
r.headers
// ^ And this will be a completely different
// headers instance constructed from "h" as init. 
```

Reply:
> Please, this would be best introduced on its own. Let's not add it as a part of this pull request.

Reverted the change

> This isn't a solution...

Got yo!

Reply:
I will look at that test today, see what I can find.